docs/runtimes/create-and-use-functions.ipynb

```diff
@@ -0,0 +1,550 @@
+    "\n",
+    "![MLRun Function](../_static/images/mlrun_concepts_architecture.png)\n",
+    "\n",
+    "Once MLRun Functions and Workflows are created and registered into the Project, they are invoked using the Project object. This workflow pairs especially well with [**Git**](../projects/use-git-manage-projects.html) and [**CI/CD**](../projects/ci-integration.html) integration."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd31f4a3",
+   "metadata": {},
+   "source": [
+    "<a id=\"creating-functions\"></a>\n",
+    "## Creating Functions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bada099b",
+   "metadata": {},
+   "source": [
+    "The recommended way to create an MLRun function is using an MLRun project (discussed in more detail in the [**Using set_function**](#using-set_function) section). The general flow looks like the following:\n",
+    "\n",
+    "```python\n",
+    "project = mlrun.get_or_create_project(...)\n",
+    "\n",
+    "fn = project.set_function(...)\n",
+    "```\n",
+    "\n",
+    "When creating a function, there are 3 main scenarios:\n",
+    "1. [**Single Source File**](#single-source-file) - when your code can be contained into a single file\n",
+    "2. [**Multiple Source Files**](#multiple-source-files) - when your code requires additional files or dependencies\n",
+    "3. [**Import Existing Function**](#import-existing-function) - when your function already exists elsewhere and you just want to import it\n",
+    "\n",
+    "``` {admonition} Note\n",
+    "Using the `set_function` method of an MLRun project allows for each of these scenarios in a transparent way. Depending on the source passed in, the project will register the function using some lower level functions. For specific use cases, you also have access to the lower level functions {py:func}`~mlrun.run.new_function`, {py:func}`~mlrun.run.code_to_function`, and {py:func}`~mlrun.run.import_function`.\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8ab1e780",
+   "metadata": {},
+   "source": [
+    "<a id=\"using-set_function\"></a>\n",
+    "### Using set_function"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c878374d",
+   "metadata": {},
+   "source": [
+    "The MLRun project object has a method called {py:meth}`~mlrun.projects.MlrunProject.set_function` which is a one-size-fits-all way of creating an MLRun function. This method accepts a variety of sources including Python files, Jupyter notebooks, Git repos, and more.\n",
+    "\n",
+    "``` {admonition} Note\n",
+    "The return value of `set_function` will be your MLRun function. You can immediately run it or apply additional configuration like resources, scaling, etc. See [**Customizing Functions**](#customizing-functions) for more details.\n",
```

Comment:
"The return value of `set_function` is your MLRun function. You can immediately run it or apply additional configuration like resources, scaling, etc. See [**Customizing functions**](#customizing-functions) for more details.\n",